File: Python/sumsquareddigits.py
'''
 *
 *	Problem: "Sum Squared Digits Function"
 *
'''

import sys

cases = int(input())

for i in range(cases):
	caseNum,base,sum = map(int, input().split(" "))
	sumSquaredDig = 0
	while (sum  > 0):
		digit = (sum % base)
		sumSquaredDig += digit**2
		sum = (sum - digit) / base
	print("%d %d" % (caseNum, sumSquaredDig))

File: Java/sumsquareddigits.java
/*
 *	Problem: "Sum Squared Digits Function"
 */

import java.util.Scanner;

public class sumsquareddigits {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int cases = sc.nextInt();
		
		while (cases-- > 0) {
			int caseNum = sc.nextInt();
			int base = sc.nextInt();
			int sum = sc.nextInt();
			
			int sumSquaredDig = 0;
			while (sum > 0) {
				int digit = (sum % base);
				sumSquaredDig += digit * digit;
				sum = (sum - digit) / base;
			}
			
			System.out.println(caseNum + " " + sumSquaredDig);
		}
	}
	
}